api/src/main/java/net/luckperms/api/platform/PluginMetadata.java

```diff
@@ -46,4 +48,11 @@
      */
     @NonNull String getApiVersion();
 
+    /**
+     * Gets the config directory path
+     *
+     * @return the directory path in where the plugin stores various configuration files
+     */
+    @NonNull Path getConfigDirectory();
```

Comment:
Why? I don't see a good usecase for this.

Reply:
My main thought was configurable extensions, it can be made in any way to be fair but this was what first came to mind.